PGO train improvements

StephanDollberg opened the pull request:
A few minor pgo train improvements:

 - pgo: Skip warmup in OMB training
 - Fix workload/driver names
 - Lower rate in OMB run
 - Forward negative exit codes
 - Check omb message count


## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.3.x
- [ ] v25.2.x
- [ ] v25.1.x

## Release Notes

* none

StephanDollberg committed:
pgo: Check omb message count

Enhance the train pgo script to check we sent the expected number of
messages. This is a basic check to confirm the OMB run was normal. We
can maybe later add some latency checks though that comes with the
problem of all static latency checks.

StephanDollberg committed:
pgo: Forward negative exit codes

Now that the dev_cluster script forwards return codes we can do the same
and exit negative if so.

StephanDollberg committed:
pgo: Lower rate in OMB run

We weren't reaching the target rate on CI so lower it to an achieable
rate. This is also required now that we check the expected message
count.

Makes no measurable difference on profile quality.

StephanDollberg committed:
pgo: Fix workload/driver names

These wrong or old copy pasta.

StephanDollberg committed:
pgo: Skip warmup in OMB training

Not needed from a profile quality and just adds another minute
needlessly.

StephanDollberg committed:
pgo: Remove dead code from train script

Leftovers from previous versions.

StephanDollberg commented:
I'll merge this in a couple days once we have had a few runs in CPD.

StephanDollberg committed:
bazel: Switch omb download to vectorized-public

OMB minimal builds are now published from CI so switch to the proper
vectorized-public url.

StephanDollberg commented:
- Further downgrade rate to 20k (m5d is slow!)
 - Update omb url to vectorized-public
 - Improve isolation

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--58328fd7-79fe-47c9-b57d-87995eee3323-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/78054">build#78054</a></summary>


| test_class                          | test_method         | test_arguments              | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                                                                                                                                                                                                                                              | test_history                                                                                                                                                     |
|:------------------------------------|:--------------------|:----------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------|
| MountUnmountIcebergTest             | test_simple_remount | {"cloud_storage_type": 1}   | integration | https://buildkite.com/redpanda/redpanda/builds/78054#019b2d92-cc7c-474b-9e83-4a224fc6a865 | FLAKY         | 7/11     | Test PASSES after retries.[No significant increase in flaky rate](https://redpandadata.atlassian.net/wiki/spaces/DEVPROD/pages/1419247661/Statistical+retry+failure+reasons#%5BinlineExtension%5DCase-C%3A-Test-passes-with-a-rate-not-significantly-worse-than-baseline-(run-passes))(baseline=0.1954, p0=0.3083, reject_threshold=0.0100. adj_baseline=0.4791, p1=0.2083, trust_threshold=0.5000) | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=MountUnmountIcebergTest&test_method=test_simple_remount        |
| WriteCachingFailureInjectionE2ETest | test_crash_all      | {"use_transactions": false} | integration | https://buildkite.com/redpanda/redpanda/builds/78054#019b2d92-cc7f-4b25-8179-53eddc139adc | FLAKY         | 10/11    | Test PASSES after retries.[No significant increase in flaky rate](https://redpandadata.atlassian.net/wiki/spaces/DEVPROD/pages/1419247661/Statistical+retry+failure+reasons#%5BinlineExtension%5DCase-C%3A-Test-passes-with-a-rate-not-significantly-worse-than-baseline-(run-passes))(baseline=0.0758, p0=1.0000, reject_threshold=0.0100. adj_baseline=0.2106, p1=0.0940, trust_threshold=0.5000) | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=WriteCachingFailureInjectionE2ETest&test_method=test_crash_all |
</details>
<!--/body-->

StephanDollberg committed:
bazel: Turn sandbox back for PGO training

Previously had issues with this but seems to work fine now so turn back
on.

StephanDollberg committed:
pgo: Move rp data dir to temp dir in PGO training

Moves the data dir also under the tmp dir for better isolation.

Move the tmp dir to /dev/shm. /tmp is actually not on tmpfs in the bazel
sandbox so disk perf can become an issue.

StephanDollberg committed:
pgo: Mute python tar warning

StephanDollberg committed:
bazel: rework train_pgo script target

Make the train_pgo.py script a tool target which depends also on the
dev_cluster.py python deps.

Like this we can make train_pgo.py also rely on deps (like pyyaml).

For other targets like minio etc. we can just make those targets of
`pgo_profile` later if needed.

StephanDollberg committed:
dev_cluster: Treat -2 RP as ok

Don't fail the dev cluster for -2 (SIGINT) return codes.

We are seeing some behaviour on CI where RP might exit with -2. I don't
fully understand why that happens on CI only but it's benign as it's the
common way to stop the dev cluster. Hence ignore it.

StephanDollberg commented on tools/pgo_bolt/train_pgo.py:
```diff
@@ -202,7 +202,7 @@ def extra_rp_tar(rp_tar: str, temp_dir: str):
     with tarfile.open(rp_tar, "r") as tar:
         for member in tar.getmembers():
-            tar.extract(member, path=extract_path)
+            tar.extract(member, path=extract_path, filter="fully_trusted")
```
It seems to be some standard python 3.14 thing:

```
DeprecationWarning: Python 3.14 will, by default, filter extracted tar archives and reject files or modify their metadata. Use the filter argument to control this behavior. explain this
```

StephanDollberg commented on tools/pgo_bolt/train_pgo.py:
```diff
@@ -161,12 +166,17 @@ async def profile(args: argparse.Namespace, tmpdir: str, redpanda_bin: str):
         if cluster_task:
             await cluster_task
 
+    if failed:
```
not logging here as the actual logs should have all the details already.

StephanDollberg commented on tools/pgo_bolt/train_pgo.py:
```diff
@@ -131,21 +132,25 @@ def check_omb(omb_target: OmbTarget):
+        return 0
     try:
         print(f"Terminating {name} (pid {proc.pid})")
         os.killpg(proc.pid, signal.SIGINT)
```
0 because everything handles SIGINT. 

Though there seems to be some race condition on CI (or something else) that I don't really understand where sometimes the dev_cluster subprocesses like RP then return with -2/SIGINT but I just pushed a commit to dev_cluster.py to ignore that as I prefer to use killpg as it's just safer.

StephanDollberg commented on tools/BUILD:
```diff
@@ -101,18 +82,33 @@ pkg_tar(
+        "pgo_bolt/train_pgo.py",
+    ],
+    data = [
+        "dev_cluster.py",
```
This is just dev_cluster.py anyway and not the :dev_cluster target

StephanDollberg committed:
pgo: Print info about profiles

StephanDollberg commented on tools/pgo_bolt/train_pgo.py:
```diff
@@ -131,21 +132,25 @@ def check_omb(omb_target: OmbTarget):
 
-async def terminate(proc: asyncio.subprocess.Process, name: str):
+async def terminate(proc: asyncio.subprocess.Process, name: str) -> int:
     if not proc:
```
Removed
